### DISCUSSION TITLE
Does Tauri Command API read all stdout data in realtime or only after the child process ends?

### DISCUSSION THREAD

**Discussion by steve1316** [tauri-apps/tauri#2828] [Q&A] [answered]
For context, I have a sort of chat log component that displays messages printed out by my Python program child process spawned by Tauri Command API. I have here a simplified form to illustrate my problem:

index.tsx
```
const command = new Command("python", "main.py")

// Attach event listeners.
command.on("close", (data) => {
  console.log("Child closed")
})
command.on("error", (error) => {
  console.log("Child had an error")
})
command.stdout.on("data", (line) => {
  console.log(`Print output from Child: \n${line}`)
})
command.stderr.on("data", (line) => {
  console.log(`Print error output from Child: \n${line}`)
})

const child = await command.spawn()
console.log("PID: ", child.pid)
```

main.py
```
import time

print("Hello!")
time.sleep(3.0)
print("Hello after 3 seconds!")
print("Good bye")
```

Both Javascript and Python code have been simplified. The actual Python code works fine except for this one issue.

The issue that I have been encountering is that stdout sends the print messages after the child process has ended. The expected behavior is that "Hello!" is console logged, wait 3 seconds, then the last 2 messages are console logged. Instead, all 3 messages are console logged at the same time after waiting 3 seconds.

So my question is, does Tauri Command API read from stdout in realtime or does it output everything in stdout after the child process ends?

I have another issue regarding getting stdout from a child process spawned by the original child process using `multiprocessing.Process` but that can wait if and when I can resolve this significant problem.

**Comment by steve1316** [comment, marked as answer]
Nevermind what I posted, the answer was that I needed to flush the print statements. That also fixed my second problem.

**Reply by illud** [reply]
Hello, what do you mean by flush?

**Reply by steve1316** [reply]
```python
print("Hello!", flush = True)
...
print("Good bye", flush = True)
```

This forces my print statements to be written to the stdout stream which acts like a text file instead of being block-buffered, allowing all of my print statements to be intercepted by the Tauri stdout listener in real time rather than waiting for its data buffer to fill up before sending them. This will then allow me to display all of those print statements onto my console-like UI element on the frontend in real time as well.

Perfect for my application use case where my users can see what the child process is doing and can provide insight for me to debug whatever issue they encounter.

**Reply by illud** [reply]
Thanks you!
